logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/CreateBackupUseCase.kt

```diff
@@ -78,32 +78,33 @@ internal class CreateBackupUseCaseImpl(
     private val idMapper: IdMapper = MapperProvider.idMapper(),
 ) : CreateBackupUseCase {
 
-    override suspend operator fun invoke(password: String): CreateBackupResult = withContext(dispatchers.default) {
-        val userHandle = userRepository.getSelfUser()?.handle?.replace(".", "-")
-        val timeStamp = DateTimeUtil.currentSimpleDateTimeString()
-        val backupName = createBackupFileName(userHandle, timeStamp)
-        val backupFilePath = kaliumFileSystem.tempFilePath(backupName)
-        deletePreviousBackupFiles(backupFilePath)
-
-        val plainDBPath =
-            databaseExporter.exportToPlainDB(securityHelper.userDBOrSecretNull(userId))?.toPath()
-                ?: return@withContext CreateBackupResult.Failure(StorageFailure.DataNotFound)
-
-        try {
-            createBackupFile(userId, plainDBPath, backupFilePath).fold(
-                { error -> CreateBackupResult.Failure(error) },
-                { (backupFilePath, backupSize) ->
-                    val isBackupEncrypted = password.isNotEmpty()
-                    if (isBackupEncrypted) {
-                        encryptAndCompressFile(backupFilePath, password)
-                    } else CreateBackupResult.Success(backupFilePath, backupSize, backupFilePath.name)
-                })
-        } finally {
-            databaseExporter.deleteBackupDBFile()
+    override suspend operator fun invoke(password: String, onProgress: (Float) -> Unit): CreateBackupResult =
+        withContext(dispatchers.default) {
+            val userHandle = userRepository.getSelfUser()?.handle?.replace(".", "-")
+            val timeStamp = DateTimeUtil.currentSimpleDateTimeString()
+            val backupName = createBackupFileName(userHandle, timeStamp)
+            val backupFilePath = kaliumFileSystem.tempFilePath(backupName)
+            deletePreviousBackupFiles(backupFilePath)
+
+            val plainDBPath =
+                databaseExporter.exportToPlainDB(securityHelper.userDBOrSecretNull(userId))?.toPath()
```

Comment:
question: Doesn't this `exportToPlainDB` also consume a long time?

I really don't know 😅.
This is the step that literally takes all the relevant content in the user's DB and exports it. Which can be a ton of data.